pkg/controller/installation/products/config/rhsso.go

```diff
@@ -0,0 +1,27 @@
+package config
+
+import "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+
+type RHSSO struct {
+	config ProductConfig
+}
+
+func newRHSSO(config ProductConfig) *RHSSO {
```

Comment:
you might be intending to address this later, but we will also need an `AddClient(name string, definition string)` and a `getClient(name string) string`, and a `SetOauthURL(url string)` and `GetOauthURL() string`.

This allows, for example 3scale, to have: 
```
rhssoConfig, err := configmanager.readRHSSO()
clientDetails := rhssoConfig.GetClient("3scale")
oauthUrl := rhssoConfig.GetOauthURL()
```
so that it can configure itself to use the cluster RHSSO

Reply:
maybe that client definition should be an struct rather than a string actually